lib/rule-tester/rule-tester.js

```diff
@@ -1246,8 +1250,9 @@ class RuleTester {
 				: null;
 
 			assert.ok(
-				ruleHasMetaMessages || requireMessage !== "messageId",
-				`Assertion options can not use 'requireMessage: "messageId"' if rule under test doesn't define 'meta.messages'.`,
+				ruleHasMetaMessages ||
+					(requireMessage !== "messageId" && !requireData),
+				`Assertion options can not use '${requireData ? "requireData" : 'requireMessage: "messageId"'}' if rule under test doesn't define 'meta.messages'.`,
```

Comment:
I think the additional check here isn't necessary, for the same reasons as in https://github.com/eslint/eslint/pull/20409#issuecomment-3689730019.